src/EntryPoints/REST/ApprovePageApi.php

```diff
@@ -29,9 +32,13 @@ public function run( int $pageId ): Response {
 			return $this->newAuthorizationFailedResponse();
 		}
 
-		// TODO: $persistence->markAsApproved( $pageId, $userId );
-		// TODO: $persistence->saveHtml( $pageId, $html );
-		// TODO: $this->newApproveFailedResponse();
+		try {
+			$this->approvalLog->approvePage( $pageId, $this->getAuthority()->getUser()->getId() );
+		} catch ( DBError $error ) {
```

Comment:
This error is not specified by our ApprovalLog interface or the DatabaseApprovalLog implementation. However, the database instance in DatabaseApprovalLog can throw such an exception and we need some way to know if the database operation, and hence the (un-)approval itself, fails.

Reply:
See my comment in the description at https://github.com/ProfessionalWiki/PageApprovals/pull/16

Same logic: why not let the exception bubble up?

If you do catch it, then it's better to catch the most generic thing, rather than only `DBError`

Reply:
I removed the exception handling.

The exception bubbles up and the API returns a 500 error code. Since there was, and likely won't be, any special response message necessary and since I was just returning 500 anyway, it seems unnecessary.

Reply:
I suspect it's then also not necessary to have `ThrowingApprovalLog` and the tests containing it.

Reply:
I removed `ThrowingApprovalLog`. The tests were basically just checking if an exception was thrown when using a test double throwing that exception.